elementary/clients/gcs/client.py

```diff
@@ -19,15 +19,23 @@ def __init__(self, config: Config):
     def create_client(cls, config: Config) -> Optional['GCSClient']:
         return cls(config) if config.has_gcs else None
 
-    def send_report(self, html_path: str) -> bool:
+    # html_path is the local path where the report generated at.
+    # bucket_file_path is where we want to upload the report to at the bucket - support folders!
+    def send_report(self, html_path: str, bucket_file_path: Optional[str] = None) -> bool:
         report_filename = path.basename(html_path)
+        bucket_report_path = bucket_file_path if bucket_file_path else report_filename
```

Comment:
here it's a bit confusing - why not saving it again to bucket_file_path?

Reply:
Override a variable that you get from the function argument is a bad practice IMO and always has the potential to cause weird behaviours.
There is the path you retrieve as an input, and there is the final report path that you actually provide to the bucket. Those are not always the same (depends on the case)